packages/core/admin/admin/src/content-manager/hooks/useKeyboardDragAndDrop.ts

```diff
@@ -1,10 +1,10 @@
 import * as React from 'react';
 
-export type UseKeyboardDragAndDropCallbacks = {
-  onCancel?: (index: number | Array<number>) => void;
-  onDropItem?: (currentIndex: number | Array<number>, newIndex?: number | Array<number>) => void;
-  onGrabItem?: (index: number | Array<number>) => void;
-  onMoveItem?: (newIndex: number | Array<number>, currentIndex: number | Array<number>) => void;
+export type UseKeyboardDragAndDropCallbacks<TIndex extends number | Array<number> = number> = {
```

Comment:
I think default value as just number wont work with Blocks, as its always going to be always number then

Reply:
No it won't be always number because this type is built through using the hook, it would only be number if in Blocks you were extending this type without passing it a generic:

```
interface NumberDefaultProps extends UseKeyboardDragAndDropCallbacks {} // index will be a number

interface NumberProps extends UseKeyboardDragAndDropCallbacks<number> {} // index will be a number

interface ArrayProps extends UseKeyboardDragAndDropCallbacks<number[]> {} // index will be an array of numbers
```

But because you're not extending this type it won't have an impact.

Reply:
yeah right!! 👍

Reply:
Great question though! ⭐